engine/src/cmd/missile.cpp

```diff
@@ -0,0 +1,364 @@
+        Unit *target = Unit::Target();
+        BOOST_LOG_TRIVIAL(info) << boost::format("Missile discharged (target %1%)")
+                                % ((target != NULL) ? target->name.get() : "NULL");
+        _Universe->activeStarSystem()->AddMissileToQueue(
+            new MissileEffect( Position(), damage, phasedamage,
+            radial_effect, radial_multiplier, owner ) );
+    discharged = true;
+  }
+}
+
+
+float Missile::ExplosionRadius()
+{
+    static float missile_multiplier =
+        XMLSupport::parse_float( vs_config->getVariable( "graphics", "missile_explosion_radius_mult", "1" ) );
+
+    return radial_effect*(missile_multiplier);
+}
+
+
+enum clsptr Missile::isUnit() const
+{
+    return MISSILEPTR;
+}
+
+
+void Missile::Kill( bool erase ) {
+    Discharge();
+    GameUnit< DummyUnit >::Kill( erase );
+}
+
+
+void Missile::UpdatePhysics2( const Transformation &trans,
+                              const Transformation &old_physical_state,
+                              const Vector &accel,
+                              float difficulty,
+                              const Matrix &transmat,
+                              const Vector &CumulativeVelocity,
+                              bool ResolveLast,
+                              UnitCollection *uc )
+{
+    // First we move the missile by calling the super
+    GameUnit<DummyUnit>::UpdatePhysics2( trans, old_physical_state, accel, difficulty, transmat, CumulativeVelocity, ResolveLast, uc );
+
+
+    // Get the target
+    Unit *target = Target();
+
+    // We have a valid target - we mark this as it affects fuel and TTL
+    if(target != nullptr)
+    {
+        had_target = true;
+    }
+
+    // Try and break lock with ECM
+    target = breakECMLock(target);
+
+    // We have a valid target
+    if(target != nullptr)
+    {
+        // But the target is dead already
+        if (target->hull < 0) {
+            target = nullptr;
+        } else {
+            // What does this do?
+            static unsigned int pointdef = ROLES::getRole( "POINTDEF" );
+            target->graphicOptions.missilelock = true;
+            un_iter i = target->getSubUnits();
+
+            Unit   *su;
+            for (; (su = *i) != nullptr; ++i)
+            {
+                if (su->attackPreference() == pointdef)
+                {
+                    if (su->Target() == nullptr)
+                    {
+                        float speed, range, mrange;
+                        su->getAverageGunSpeed( speed, range, mrange );
+                        if ( ( Position()-su->Position() ).MagnitudeSquared() < range*range )
+                        {
+                            su->Target( this );
+                            su->TargetTurret( this );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Go wild
+    // I'm pretty sure this code does nothing so I'm commenting it out
```

Comment:
Do we **know** for certain that this code does nothing? If not, it might be a good idea to uncomment it.

Reply:
As I said, I'm not completely sure, but as I refactored everything around it and merged two codes together, it can't stay anyway. 
In any case, it's a re-targeting code, so at worst it'll be a niche bug. After we finish refactoring other stuff, we can look into missiles without lock, etc. 
The current in-game has heat seeking vs. image recognition (IR) and I'm pretty sure it isn't implemented that way. That is, if we break the lock, the IR missile will pick another target with the same silhouette.

Reply:
@royfalk I'm good with that; just record which issue will track that long term.

Reply:
#328 will track this.